### PR TITLE
Adding wiper signals

### PR DESCRIPTION
**Pull request by erikbosch**
Idea:

Define two set of signals - usercontrol and systemcontrol

### CODE REVIEW

File: spec/Body/Body.vspec [hunk 1]
```diff
@@ -90,11 +90,102 @@ Windshield.Wiping:
...
+  datatype: boolean
+  type: actuator
+  description: Specifies behavior if communication is lost.
+               True = Fail Safe Mode Active, wipers may e.g. start automatically if communication is lost.
+               False = False Safe Mode Inactive.
+
+
+Windshield.Wiping.SystemControl.Load:
+  datatype: uint16
+  unit: W
```
Comment (erikbosch) [inline spec/Body/Body.vspec]: Watt currently not supported - fixed by https://github.com/COVESA/vss-tools/pull/134

File: spec/Body/Body.vspec [hunk 2]
```diff
@@ -90,11 +90,102 @@ Windshield.Wiping:
...
+####################################################################
+
+Windshield.Wiping.UserControl:
+  type: branch
+  description: High Level signals typically available to user/driver to control wiper functionality
+  
+Windshield.Wiping.UserControl.RequestedMode:
   datatype: string
   type: actuator
   allowed: ["off", "slow", "medium", "fast", "interval", "rainsensor"]
```
Comment (SebastianSchildt) [inline spec/Body/Body.vspec]: I am not sure I understand what "interval" means

File: spec/Body/Body.vspec [hunk 3]
```diff
@@ -90,11 +90,102 @@ Windshield.Wiping:
...
-  description: Wiper status.
+  description: Wiper mode requested by user/driver.
+  
+Windshield.Wiping.UserControl.RequestedIntensity:
```
Comment (SebastianSchildt) [inline spec/Body/Body.vspec]: So this is combined only with settings interval or sensor?

File: spec/Body/Body.vspec [hunk 4]
```diff
@@ -90,11 +90,102 @@ Windshield.Wiping:
...
+  description: Low Level signals to control behavior of wiper system in detail.
+               These signals are typically not directly available to the user/driver of the vehicle.
+
+
+Windshield.Wiping.SystemControl.Mode:
+  datatype: string
+  type: actuator
+  allowed: [
+    STOP, # Wipers finish current wiping cycle and then moves to parking position
+    STOP_APP, # Wipers finish current wiping cycle and then moves to alternative parking position
```
Comment (SebastianSchildt) [inline spec/Body/Body.vspec]: We probably need to explain that APP is done to lessen wear of the wipers, otherwise it is unclear to 99% of the population

**Review by SebastianSchildt — COMMENTED** [review]
Just some comments.

We might also need something "Wiper worn" or "Needs replacement" , we should see we can follow modelling from e.g. brake pads? Maybe we need "Wiping.Service",

but that would sort of imply a new pattern in VSS: UserControl/SystemControl/Service. So we should think hard, if this really makes sense (seems good to me on first sight)

File: spec/Body/Body.vspec (hunk 2 above)
Comment (erikbosch) [reply spec/Body/Body.vspec]: My interpretation is that this is the functionality where the wiper do one cycle, and then wait X seconds before it does the next cycle, then wait X seconds and so on. Quite common solution on cars without rain sensor.

File: spec/Body/Body.vspec (hunk 3 above)
Comment (erikbosch) [reply spec/Body/Body.vspec]: Yes, this is an example from my side that would cover the two cars we currently own (one from 2000, one from 2017). The older one has one handle where you can select "single wipe", "interval", "slow, or "fast", and a wheel on the handle with around 4 different positions for interval. The newer one has one handle where you can select "single wipe", "rain sensor", "slow, or "fast", and a wheel on the handle with around 4 different positions for rain sensor sensitivity/intentisty..

**Comment by erikbosch** [comment]
> but that would sort of imply a new pattern in VSS: UserControl/SystemControl/Service. So we should think hard, if this really makes sense (seems good to me on first sight)

Could be a good pattern

**Comment by erikbosch** [comment]
Handled by https://github.com/COVESA/vehicle_signal_specification/pull/431 and https://github.com/COVESA/vehicle_signal_specification/pull/454

**Pull request closed without merging**